### PR TITLE
feat: implement junit xml reporter

### PR DESCRIPTION
**Pull request by thesayyn**

### CODE REVIEW

File: jasmine/private/jasmine_test.bzl [hunk 1]
```diff
@@ -6,26 +6,36 @@ _attrs = dicts.add(js_binary_lib.attrs, {
...
         include_npm_linked_packages = ctx.attr.include_npm_linked_packages,
     )
     files.extend(ctx.files.data)
+    files.append(ctx.file.junit_reporter)
+
+
+    junit_reporter = ctx.file.junit_reporter.short_path
+
+    if ctx.attr.chdir:
+        junit_reporter = "/".join([".." for _ in ctx.attr.chdir.split("/")] + [junit_reporter])
```
Comment (jbedard) [inline jasmine/private/jasmine_test.bzl:33]: Can `chdir` not have things like `.` or `..` in it? Can't we use something other then `.short_path`?

File: jasmine/private/junit_reporter.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const reporters = require("jasmine-reporters");
+const path = require("path");
+
+const xmlOutputFile = process.env.XML_OUTPUT_FILE;
+
+jasmine.getEnv().addReporter(
```
Comment (jbedard) [inline jasmine/private/junit_reporter.js:6]: I assume `jasmine` is a global but is it possible to `require()` it instead? 🤷

**Comment by jbedard** [comment]
How could we add a test for this? 🤔

**Review by jbedard — APPROVED** [review]

File: jasmine/private/jasmine_test.bzl (hunk 1 above)
Comment (thesayyn) [reply jasmine/private/jasmine_test.bzl:33]: short_part already gives us what we want. the pwd is `$RUNFILES_DIR/WKSP` and the short path is `../jasmine/junit_reporter.js`. chdir doesn't allow those characters as far as I know.

File: jasmine/private/junit_reporter.js (hunk 2 above)
Comment (thesayyn) [reply jasmine/private/junit_reporter.js:6]: this is the recommended pattern by jasmine. if a require jasmine, it'll give me the library itself not the instance of it. thus I can't do getEnv() on an uninitialized instance.

Comment (thesayyn) [reply jasmine/private/junit_reporter.js:6]: I'll make sure that this is the case though.

File: jasmine/private/jasmine_test.bzl (hunk 1 above)
Comment (thesayyn) [reply jasmine/private/jasmine_test.bzl:33]: I believe my options are either using a runfiles helper or processing this argument in runner.js like js_binary does in it's launcher. maybe @gregmagolan knows more?

**Comment by thesayyn** [comment]
> How could we add a test for this? 🤔

added one.

File: jasmine/private/junit_reporter.js (hunk 2 above)
Comment (thesayyn) [reply jasmine/private/junit_reporter.js:6]: yeap. that's the case. i get `TypeError: require(...).getEnv is not a function`

**Pull request merged**